contracts/RolesRegistry/SftRolesRegistrySingleRole.sol

```diff
@@ -45,118 +32,78 @@ contract SftRolesRegistrySingleRole is ISftRolesRegistry, ERC1155Holder {
         _;
     }
 
-    modifier validRoleAndGrantee(
-        address _grantor,
+    modifier sameGrantee(
+        uint256 _recordId,
         bytes32 _role,
-        address _grantee,
-        uint256 _nonce
+        address _grantee
     ) {
-        require(_role == UNIQUE_ROLE, 'SftRolesRegistry: role not supported');
         require(
-            _grantee != address(0) && _grantee == roleAssignments[_grantor][_nonce][_role].grantee,
+            _grantee != address(0) && _grantee == roleAssignments[_recordId][_role].grantee,
             'SftRolesRegistry: grantee mismatch'
         );
         _;
     }
 
     /** External Functions **/
 
-    function grantRoleFrom(
-        RoleAssignment calldata _grantRoleData
-    )
-        external
-        override
-        validGrantRoleData(
-            _grantRoleData.nonce,
-            _grantRoleData.grantee,
-            _grantRoleData.expirationDate,
-            _grantRoleData.tokenAmount,
-            _grantRoleData.role
-        )
-        onlyOwnerOrApproved(_grantRoleData.grantor, _grantRoleData.tokenAddress)
-    {
-        if (deposits[_grantRoleData.grantor][_grantRoleData.nonce].tokenAmount == 0) {
-            // nonce does not exist, transfer tokens
-            _deposit(_grantRoleData);
-        } else {
-            // nonce exists
-            require(
-                deposits[_grantRoleData.grantor][_grantRoleData.nonce].tokenAddress == _grantRoleData.tokenAddress,
-                'SftRolesRegistry: tokenAddress mismatch'
-            );
-            require(
-                deposits[_grantRoleData.grantor][_grantRoleData.nonce].tokenId == _grantRoleData.tokenId,
-                'SftRolesRegistry: tokenId mismatch'
-            );
-            require(
-                deposits[_grantRoleData.grantor][_grantRoleData.nonce].tokenAmount == _grantRoleData.tokenAmount,
-                'SftRolesRegistry: tokenAmount mismatch'
-            );
-
-            RoleData storage _roleData = roleAssignments[_grantRoleData.grantor][_grantRoleData.nonce][
-                _grantRoleData.role
-            ];
-            require(
-                _roleData.expirationDate < block.timestamp || _roleData.revocable,
-                'SftRolesRegistry: nonce is not expired or is not revocable'
-            );
-        }
-        _grantOrUpdateRole(_grantRoleData);
+    function createRecordFrom(
+        address _grantor,
+        address _tokenAddress,
+        uint256 _tokenId,
+        uint256 _tokenAmount
+    ) external override onlyOwnerOrApproved(_grantor, _tokenAddress) returns (uint256 recordId_) {
+        require(_tokenAmount > 0, 'SftRolesRegistry: tokenAmount must be greater than zero');
+        recordId_ = _createRecord(_grantor, _tokenAddress, _tokenId, _tokenAmount);
     }
 
-    function revokeRoleFrom(
-        address _grantor,
-        uint256 _nonce,
+    function grantRole(
```

Comment:
since this function has the onlyOwnerOrApproved modifier. I think it should add `From` sufix.
```suggestion
    function grantRoleFrom(
```

Reply:
I considered it initially, but ultimately decided not to include `From`. In EIPs such as ERC-721, `transferFrom` includes anan `address _from` as parameter, which is not the case here. The `recordId` is an entity which can be accessed by different users (grantor, approved users, and grantee), no need to pass `_from`